docs/samples/Microsoft.ML.Samples/Dynamic/ValueMapping.cs

```diff
@@ -0,0 +1,90 @@
+            // as well as the source of randomness.
+            var ml = new MLContext();
+
+            // Get a small dataset as an IEnumerable.
+            IEnumerable<SamplesUtils.DatasetUtils.SampleInfertData> data = SamplesUtils.DatasetUtils.GetInfertData();
+            var trainData = ml.Data.ReadFromEnumerable(data);
+
+            // Preview of the data.
+            //
+            // Age    Case  Education  induced     parity  pooled.stratum  row_num  ...
+            // 26.0   1.0   0-5yrs      1.0         6.0       3.0      1.0  ...
+            // 42.0   1.0   0-5yrs      1.0         1.0       1.0      2.0  ...
+            // 39.0   1.0   12+yrs      2.0         6.0       4.0      3.0  ...
+            // 34.0   1.0   0-5yrs      2.0         4.0       2.0      4.0  ...
+            // 35.0   1.0   6-11yrs     1.0         3.0       32.0     5.0  ...
+
+            // Creating a list of keys based on the Education values from the dataset
+            // These lists are created by hand for the demonstration, but the ValueMappingEstimator does take an IEnumerable.
+            var educationKeys = new List<string>()
+            {
+                "0-5yrs",
+                "6-11yrs",
+                "12+yrs"
+            };
+
+            // Creating a list of associated values that will map respectively to each educationKey
+            var educationValues = new List<string>()
+            {
+                "Cat1",
+                "Cat2", 
+                "Cat3"
+            };
+
+            // Constructs the ValueMappingEstimator making the ML.net pipeline
+            var pipeline = new ValueMappingEstimator<string, string>(ml, educationKeys, educationValues, ("EducationCategory", "Education"));
+
+            // Fits the ValueMappingEstimator and transforms the data converting the Education to EducationCategory.
+            IDataView transformedData = pipeline.Fit(trainData).Transform(trainData);
+
+            // Getting the resulting data as an IEnumerable of SampleInfertDataWithFeatures. This will contain the newly created column EducationCategory
+            IEnumerable<SampleInfertDataWithFeatures> featureRows = ml.CreateEnumerable<SampleInfertDataWithFeatures>(transformedData, reuseRowObject: false);
+            
+            Console.WriteLine($"Example of mapping string->string");
+            Console.WriteLine($"Age\tEducation\tEducationLabel");
+            foreach (var featureRow in featureRows)
+            {
+                Console.WriteLine($"{featureRow.Age}\t{featureRow.Education}  \t{featureRow.EducationCategory}");
+            }
+
+            // Features column obtained post-transformation.
+            //
+            // Age Education    EducationLabel
+            // 26  0-5yrs       Cat1
+            // 42  0-5yrs       Cat1
+            // 39  12+yrs       Cat3
+            // 34  0-5yrs       Cat1
+            // 35  6-11yrs      Cat2
```

Comment:
Related to my comment above, what is the use  of changing it from a string to a different string? Can you show what this can then be used for?

Reply:
Added a remarks section and added additional verbage to each example.

---
In reply to: [252778545](https://github.com/dotnet/machinelearning/pull/2222#discussion_r252778545) [](ancestors = 252778545)